synapse/handlers/sync.py

```diff
@@ -636,7 +636,7 @@ async def _load_filtered_recents(
         # Don't bother to bundle aggregations if the timeline is unlimited,
         # as clients will have all the necessary information.
         bundled_aggregations = None
-        if limited:
```

Comment:
I wonder if these are equivalent actually, see https://github.com/matrix-org/synapse/pull/11659/commits/acdc04a897a04f20033396652016d563cbf273a3#diff-70c84622edff898486bdc02d7a579db4e52a3e0b26c14020bed0ca1fb8ea2018L519-L524

Reply:
TL;DR:
- sync code is confusing
- I think this is fine as it stands

Reply:
Yes, it gets flipped a bit. Would it be clearer to set it again once to `limited or newly_joined_room` and use that twice?

Reply:
IMO leave it as is, in favour of another sync code tidy up to clarify what `limited` means and all the various bits of flipping that go on.

Reply:
Makes sense to me. Thanks!